ownership-chain/runtime/src/tests/xcm_tests.rs

```diff
@@ -0,0 +1,165 @@
+//! Cross-chain asset transfer tests
+
+use super::xcm_mock::{parachain::Assets, MockNet, ParaA, ParaB, Relay};
+use crate::{
+	tests::xcm_mock::{
+		msg_queue::mock_msg_queue, parachain, ParachainBalances, ParachainXtokens,
+		RelayChainPalletXcm, ALITH, BOBTH, INITIAL_BALANCE,
+	},
+	UNIT,
+};
+use cumulus_primitives_core::{
+	Fungibility::Fungible,
+	Instruction::{BuyExecution, Transact, WithdrawAsset},
+	Junction::{AccountKey20, Parachain},
+	Junctions::{Here, X1, X2},
+	MultiAsset, MultiLocation, NetworkId, OriginKind,
+	WeightLimit::Unlimited,
+	Xcm,
+};
+use frame_support::{assert_ok, traits::fungibles::Inspect, weights::Weight};
+use parity_scale_codec::Encode;
+use xcm_simulator::TestExt;
+
+/// Test downward message passing. Does some basic remark in the parachain from the relay chain.
+#[test]
+fn basic_dmp() {
+	MockNet::reset();
+
+	let remark = parachain::RuntimeCall::System(
+		frame_system::Call::<parachain::Runtime>::remark_with_event { remark: vec![1, 2, 3] },
+	);
+
+	Relay::execute_with(|| {
+		assert_ok!(RelayChainPalletXcm::send_xcm(
+			Here,
+			Parachain(1),
+			Xcm(vec![
+				WithdrawAsset(
+					vec![MultiAsset {
+						id: MultiLocation::here().into(),
+						fun: cumulus_primitives_core::Fungibility::Fungible(UNIT)
+					}]
+					.into()
+				),
+				BuyExecution {
+					fees: MultiAsset {
+						id: MultiLocation::here().into(),
+						fun: cumulus_primitives_core::Fungibility::Fungible(UNIT)
+					},
+					weight_limit: Unlimited,
+				},
+				Transact {
+					origin_kind: OriginKind::SovereignAccount,
+					require_weight_at_most: Weight::from_parts(1_000_000_000, 1024 * 1024),
+					call: remark.encode().into(),
+				}
+			]),
+		));
+	});
+
+	// Execute remote transact and verify that `Remarked` event is emitted.
+	ParaA::execute_with(|| {
+		use parachain::{RuntimeEvent, System};
+		assert!(System::events().iter().any(|r| matches!(
+			r.event,
+			RuntimeEvent::System(frame_system::Event::Remarked { .. })
+		)));
+	});
+}
+
+#[test]
+fn para_to_para_native_transfer_and_back() {
+	MockNet::reset();
+
+	// in ParaB, we need to set up and register the token of ParaA
+	ParaB::execute_with(|| {
+		// we only need to create the asset, registering location and setting unit price
+		// per second is not necessary, since it is already mocked by
+		assert_ok!(Assets::force_create(
+			parachain::RuntimeOrigin::root(),
+			2_u32,
+			ALITH,
+			false,
+			UNIT
+		));
+	});
+
+	let amount = 100 * UNIT;
+	ParaA::execute_with(|| {
+		// bob in ParaB
+		let destination = MultiLocation {
+			parents: 1,
+			interior: X2(
+				Parachain(2),
+				AccountKey20 { network: Some(NetworkId::Kusama), key: BOBTH.0 },
+			),
+		};
+
+		assert_ok!(ParachainXtokens::transfer(
```

Comment:
In a real scenario (KLAOS => Astar) we don't have `xtokens` pallets, I suggest use `reserveTransferAsset` from `pallet-xcm` instead

Reply:
this one makes sense. I also felt a bit uneasy because I am using `xtokens` and `assets` pallets, even though the real runtime does not. 

I can refactor to not use `xtokens` at all, but `assets` would be required (I guess it would be a mock of a minimal runtime that parachain should have to support foreign assets, etc.).

Reply:
I see your point but yeah I think it might be worth try to have the closest environment to the real world until we integrate emulator/chopstick. I think it is right to have `assets` as in the real world Astar does have it.